syne_tune/optimizer/schedulers/transfer_learning/rush.py

```diff
@@ -112,7 +112,8 @@ def __setstate__(self, state):
         self._thresholds = pickle.loads(state['thresholds'])
 
     def _on_trial_result(self, trial_decision: str, trial: Trial, result: Dict) -> str:
-        if trial_decision == SchedulerDecision.STOP:
+        if trial_decision == SchedulerDecision.STOP or \
+                trial_decision == SchedulerDecision.PAUSE and self._hyperband_scheduler.scheduler_type == 'stopping':
```

Comment:
This does not fix the issues.

In type='promotion', this happens:
- A trial is always paused at the next milestone (if rung levels are 1, 3, 9, ...), a trial promoted from 1 has milestone 3, so the next rung level)
- The interesting decision is which trials get promoted, and that happens in _suggest

I bet that RUSH needs that decision to be influenced.

Reply:
@wistuba : Perhaps we can only support "stop" type for now?

Reply:
> This does not fix the issues.
> 
> In type='promotion', this happens:
> 
>     * A trial is always paused at the next milestone (if rung levels are 1, 3, 9, ...), a trial promoted from 1 has milestone 3, so the next rung level)
> 
>     * The interesting decision is which trials get promoted, and that happens in _suggest
> 
> 
> I bet that RUSH needs that decision to be influenced.

The new piece of code is actually L133. At this point we identify for scheduler type 'promotion' that a milestone is reached and decide whether to stop or not. Does that not work?

Reply:
test_given_trial_not_meeting_threshold_return_stop shows that we correctly chose STOP rather than PAUSE. Not sure whether this is sufficient to make sure the trial is never suggested again.

Reply:
It would be OK with me to only support stopping, but I am not sure this is so good practice in the long run. How to communicate this to a user? Also, I suppose the overall task of "RUSH in Syne Tune" would not be done.

Reply:
I cant comment on the above question of Martin whether this behavior will be correct to support promotion.

On the other point, from my side it is really fine to support only stopping type. I would still mark it as done, the initial paper does not consider at all the async case, having one of those mode is fine in my view given that they dont make such a big impact in final results.